### ISSUE TITLE
Wagtail userbar is not accessible

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#6108]
### Issue Summary

The current implementation of the Wagtail userbar (the Wagtail circle button that appears front facing on pages when signed in) is not implemented in an accessible way.

### Steps to Reproduce

1. Load up a Wagtail project and sign in
2. Go to the home page (published)
3. Attempt to use a screen reader or keyboard only to navigate to the userbar
4. Expected: Menu should be activated with `space`, `enter` and then focus should move to the menu items (either the first item or the container), screen readers should be advised that this controls a menu.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

<img width="1436" alt="wagtail-userbar-firefox-accessibility-2020-06-04" src="https://user-images.githubusercontent.com/1396140/83689639-b72d5b00-a632-11ea-9587-28d9309441e6.png">

### Technical details

* Python version: 3.6.9
* Django version: 3.0.5
* Wagtail version: 2.10 (pre-release)
* Browser version: Firefox 76 on Mac 10.15.4

### Potential Solution

* At minimum, the trigger in the [userbar base template](https://github.com/wagtail/wagtail/blob/master/wagtail/admin/templates/wagtailadmin/userbar/base.html) should be made into an actual DOM button, which might even remove the need for the custom touch logic in [userbar.js](https://github.com/wagtail/wagtail/blob/master/wagtail/admin/static_src/wagtailadmin/js/userbar.js)
* Ideally the implementation should follow a best practice for nav like popover menus (some links below)
* It might be good to move the trigger to after the menu items, meaning it will be focused after tabbing through the links, not a full focus management solution but enough to make it work well without too much extra JS.
* https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-actions.html
* https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html
* https://www.w3.org/WAI/tutorials/menus/flyout/

**Comment by lb-** [comment]
I have made a start, still working through some issues, but if I do not get to finish this here it is
https://github.com/lb-/wagtail/commit/bd30c966564a927f7e66e777fcc8678ecc7b5e7f
